src/widgets/modals/PatientEdit.js

```diff
@@ -0,0 +1,265 @@
+            <ValidationTextInput
+              value={addressTwo}
+              label="Address 2"
+              invalidMessage="Must be less than 50 characters"
+              onValidate={validateAddress}
+              onChangeValidState={setAddressTwoValidity}
+              onSubmitEditing={setAddressTwo}
+            />
+            <ValidationTextInput
+              value={country}
+              label="Country"
+              invalidMessage="Must be less than 20 characters"
+              onValidate={validateCountry}
+              onChangeValidState={setCountryValidity}
+              onSubmitEditing={setCountry}
+            />
+          </View>
+          <View style={flexOne} />
+        </View>
+      </ScrollView>
+      <View style={buttonsRow}>
+        <PageButton
+          style={saveButton}
+          isDisabled={!canSave}
+          textStyle={saveButtonTextStyle}
+          text="Save"
+          onPress={savePatient}
+        />
+        <PageButton style={cancelButton} textStyle={cancelButtonTextStyle} text="Cancel" />
+      </View>
+    </View>
+  );
+};
+
+const mapDispatchToProps = dispatch => ({
+  setDateOfBirthValidity: newValidity =>
+    dispatch(PatientActions.setFieldValidity('dateOfBirth', newValidity)),
+  setAddressOneValidity: newValidity =>
+    dispatch(PatientActions.setFieldValidity('address1', newValidity)),
+  setAddressTwoValidity: newValidity =>
+    dispatch(PatientActions.setFieldValidity('address2', newValidity)),
+  setPatientCodeValidity: newValidity =>
+    dispatch(PatientActions.setFieldValidity('code', newValidity)),
+  setCountryValidity: newValidity =>
+    dispatch(PatientActions.setFieldValidity('country', newValidity)),
+  setFirstName: newValue => dispatch(PatientActions.setFieldUpdate('firstName', newValue)),
+  setLastName: newValue => dispatch(PatientActions.setFieldUpdate('lastName', newValue)),
+  setCode: newValue => dispatch(PatientActions.setFieldUpdate('code', newValue)),
+  setDateOfBirth: newValue => dispatch(PatientActions.setFieldUpdate('dateOfBirth', newValue)),
+  setEmail: newValue => dispatch(PatientActions.setFieldUpdate('email', newValue)),
+  setPhone: newValue => dispatch(PatientActions.setFieldUpdate('phone', newValue)),
+  setAddressOne: newValue => dispatch(PatientActions.setFieldUpdate('address1', newValue)),
+  setAddressTwo: newValue => dispatch(PatientActions.setFieldUpdate('address2', newValue)),
+  setCountry: newValue => dispatch(PatientActions.setFieldUpdate('country', newValue)),
+  savePatient: () => dispatch(PatientActions.patientUpdate()),
+});
+
+const mapStateToProps = state => {
+  const { patient } = state;
+  const {
+    firstNameIsValid,
+    lastNameIsValid,
+    codeIsValid,
+    dateOfBirthIsValid,
+    phoneIsValid,
+    countryIsValid,
+    firstName,
+    lastName,
+    code,
+    dateOfBirth,
+    email,
+    phone,
+    addressOne,
+    addressTwo,
+    country,
+  } = patient;
+
+  const canSave =
+    firstNameIsValid &&
```

Comment:
I'm thinking that if `patient` did have an `isValid` property that contained a bunch of boolean props.. then there'd be a way to make this a little more generic i.e. `Object.values(patient.isValid).every(v=>v);`

Reply:
079307c